experimental/dds/tree2/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -263,23 +258,42 @@ export function isNoopMark(mark: Mark<unknown>): mark is NoopMark {
 	return mark.type === NoopMarkType;
 }
 
-export function getOffsetAtRevision(
+/**
+ * @returns The number of cells in the range which come before the position described by `lineage`.
+ */
+export function getOffsetInCellRange(
 	lineage: LineageEvent[] | undefined,
-	reattachRevision: RevisionTag | undefined,
+	revision: RevisionTag | undefined,
+	id: ChangesetLocalId,
+	count: number,
 ): number | undefined {
-	if (lineage === undefined || reattachRevision === undefined) {
+	if (lineage === undefined || revision === undefined) {
 		return undefined;
 	}
 
 	for (const event of lineage) {
-		if (event.revision === reattachRevision) {
-			return event.offset;
+		if (
+			event.revision === revision &&
+			areOverlappingIdRanges(id, count, event.id, event.count)
+		) {
+			return (event.id as number) + event.offset - id;
 		}
 	}
 
 	return undefined;
 }
 
+export function areOverlappingIdRanges(
+	id1: ChangesetLocalId,
+	count1: number,
+	id2: ChangesetLocalId,
+	count2: number,
+): boolean {
+	const lastId1 = (id1 as number) + count1 - 1;
+	const lastId2 = (id2 as number) + count2 - 1;
+	return (id1 >= id2 && id1 <= lastId2) || (id2 >= id1 && id2 <= lastId1);
```

Comment:
Nit: is it just me that finds this dramatically easier to read?

```suggestion
	return (id2 <= id1 && id1 <= lastId2) || (id1 <= id2 && id2 <= lastId1);
```

Reply:
I also prefer it. I wish more languages supported a syntax like `id2 <= id1 <= lastId2`. I think I've only seen that in Python.